### PR TITLE
Revert "[SPARK-23264][SQL] Make INTERVAL keyword optional when ANSI enabled"

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

Revert https://github.com/apache/spark/pull/20433 .
### Why are the changes needed?

According to the SQL standard, the INTERVAL prefix is required:
```
<interval literal> ::=
  INTERVAL [ <sign> ] <interval string> <interval qualifier>

<interval string> ::=
  <quote> <unquoted interval string> <quote>
```

### Does this PR introduce any user-facing change?

yes, but omitting the INTERVAL prefix is a new feature in 3.0

### How was this patch tested?

existing tests

### CODE REVIEW

File: docs/sql-keywords.md [hunk 1]
```diff
@@ -87,8 +87,7 @@ Below is a list of all the keywords in Spark SQL.
   <tr><td>DATA</td><td>non-reserved</td><td>non-reserved</td><td>non-reserved</td></tr>
   <tr><td>DATABASE</td><td>non-reserved</td><td>non-reserved</td><td>non-reserved</td></tr>
   <tr><td>DATABASES</td><td>non-reserved</td><td>non-reserved</td><td>non-reserved</td></tr>
-  <tr><td>DAY</td><td>reserved</td><td>non-reserved</td><td>reserved</td></tr>
-  <tr><td>DAYS</td><td>non-reserved</td><td>non-reserved</td><td>non-reserved</td></tr>
```
Comment (cloud-fan) [inline docs/sql-keywords.md]: according to https://www.postgresql.org/docs/current/sql-keywords-appendix.html , `unit-S` is not keyword.

File: docs/sql-keywords.md [hunk 2]
```diff
@@ -174,15 +172,9 @@ Below is a list of all the keywords in Spark SQL.
   <tr><td>MAP</td><td>non-reserved</td><td>non-reserved</td><td>non-reserved</td></tr>
   <tr><td>MATCHED</td><td>non-reserved</td><td>non-reserved</td><td>non-reserved</td></tr>
   <tr><td>MERGE</td><td>non-reserved</td><td>non-reserved</td><td>non-reserved</td></tr>
-  <tr><td>MICROSECOND</td><td>non-reserved</td><td>non-reserved</td><td>non-reserved</td></tr>
-  <tr><td>MICROSECONDS</td><td>non-reserved</td><td>non-reserved</td><td>non-reserved</td></tr>
-  <tr><td>MILLISECOND</td><td>non-reserved</td><td>non-reserved</td><td>non-reserved</td></tr>
-  <tr><td>MILLISECONDS</td><td>non-reserved</td><td>non-reserved</td><td>non-reserved</td></tr>
```
Comment (cloud-fan) [inline docs/sql-keywords.md]: according to https://www.postgresql.org/docs/current/sql-keywords-appendix.html , they are not keyword.

File: docs/sql-keywords.md [hunk 3]
```diff
@@ -293,12 +284,9 @@ Below is a list of all the keywords in Spark SQL.
   <tr><td>USING</td><td>reserved</td><td>strict-non-reserved</td><td>reserved</td></tr>
   <tr><td>VALUES</td><td>non-reserved</td><td>non-reserved</td><td>reserved</td></tr>
   <tr><td>VIEW</td><td>non-reserved</td><td>non-reserved</td><td>non-reserved</td></tr>
-  <tr><td>WEEK</td><td>non-reserved</td><td>non-reserved</td><td>non-reserved</td></tr>
```
Comment (cloud-fan) [inline docs/sql-keywords.md]: according to https://www.postgresql.org/docs/current/sql-keywords-appendix.html , this is not keyword.

**Comment by cloud-fan** [comment]
cc @maropu @gatorsmile

**Review by maropu — APPROVED** [review]
Looks ok to me if the tests passed.

File: sql/core/src/test/resources/sql-tests/inputs/ansi/interval.sql [hunk 4]
```diff
@@ -1,17 +1 @@
...
-
--- malformed interval literal with ansi mode
-select 1 year to month;
-select '1' year to second;
-select 1 year '2-1' year to month;
-select (-30) day;
-select (a + 1) day;
-select 30 day day day;
\ No newline at end of file
+--IMPORT interval.sql
```
Comment (maropu) [inline sql/core/src/test/resources/sql-tests/inputs/ansi/interval.sql]: Still need this test file?

Comment (cloud-fan) [reply sql/core/src/test/resources/sql-tests/inputs/ansi/interval.sql]: yea, the overflow behavior is different.

**Review by HyukjinKwon — APPROVED** [review]
Looks fine if tests pass

**Comment by yaooqinn** [comment]
LGTM

**Comment by cloud-fan** [comment]
retest this please

**Review by viirya — APPROVED** [review]

File: sql/core/src/test/resources/sql-tests/inputs/ansi/interval.sql (hunk 4 above)
Comment (gatorsmile) [reply sql/core/src/test/resources/sql-tests/inputs/ansi/interval.sql]: In general, we should have a comment to explain why we need this file.

**Comment by gatorsmile** [comment]
Thanks! Merged to master.

**Pull request closed without merging**